packages/dev/loaders/src/glTF/glTFFileLoader.ts

```diff
@@ -562,12 +562,26 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
      */
     public loadFile(
         scene: Scene,
-        fileOrUrl: File | string,
+        fileOrUrl: File | string | ArrayBufferView,
         onSuccess: (data: any, responseURL?: string) => void,
         onProgress?: (ev: ISceneLoaderProgressEvent) => void,
         useArrayBuffer?: boolean,
-        onError?: (request?: WebRequest, exception?: LoadFileError) => void
+        onError?: (request?: WebRequest, exception?: LoadFileError) => void,
+        name?: string
     ): IFileRequest {
+
+        if(fileOrUrl as ArrayBufferView)
+        {
+            this._loadBinary(scene, fileOrUrl as ArrayBufferView, onSuccess, onError, name);
+            
+            const fileRequest: IFileRequest = {
+                abort: () => {},
+                onCompleteObservable: new Observable<IFileRequest>(),
+            };
+            
+            return fileRequest;
+        }
+
         this._progressCallback = onProgress;
 
         const rootUrl = (fileOrUrl as File).name ? "file:" : Tools.GetFolderPath(fileOrUrl as string);

```

Comment:
I think I did this somewhat incorrectly here. `rootUrl` needs to come from SceneLoader or it may get the wrong `rootUrl` if it's passed in to the SceneLoader as opposed to derived from the url.

Reply:
I've updated loadFile to receive the rootUrl from SceneLoader